src/sql/workbench/browser/designer/designer.ts

```diff
@@ -904,11 +976,25 @@ export class Designer extends Disposable implements IThemable {
 					table.registerPlugin(deleteRowColumn);
 					columns.push(deleteRowColumn.definition);
 				}
-				table.columns = columns;
+				table.columns = buttonColumn.concat(columns);
+				table.columns.forEach((v, i) => table.columns[i].id = table.columns[i].id || i.toString());
 				table.grid.onBeforeEditCell.subscribe((e, data): boolean => {
 					return data.item[data.column.field].enabled !== false;
 				});
-
+				let disabledActions = this._actions.filter(b => b.enabled === false);
+				table.grid.onClick.subscribe((e, data) => {
+					const dataLength = table.grid.getDataLength();
+					disabledActions.forEach((b) => {
```

Comment:
1. the array should not be called disabledActions, it should be all actions.
2. the logic of whether an action is enabled/disabled should be defined it the action itself instead of here. we should call action.updateEnabledState() here.

Reply:
1. This is actually the disabled actions only, not all actions because we don't want to disable all the actions (e.g. - Add row button is enabled when there's no focus in the grid, but the Move buttons are disabled)
2. `.enable = true` is a setter from the action. What do you mean by using action.updateEnabledState()?

Reply:
the code currently works because you set the initial value to false, the actual meaning of 'disabledActions' variable is 'actions whose state could change'.

my point is you can let the actions decide their own state according to the context. 

`
foreach(action in actions) {
    action.updateState(table);
}
`
having to know all the action types doesn't seem to be a good idea. you might need to introduce a new base class/interface with updateState(table) method.